rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -526,102 +616,193 @@ def slice_state_history(
             omit_unset_slots: If `True` do not include the initial values of slots.
 
         Returns:
-            A tuple of list of states, list of actions and list of entity data.
+            A tuple of list of states, list of labels and list of entity data.
+        """
+        rasa.shared.utils.io.raise_deprecation_warning(
+            "'training_states_actions_and_entities' is being deprecated in "
+            "favor of 'training_states_labels_and_entities'."
+        )
+        return self.training_states_labels_and_entities(
+            trackers,
+            domain,
+            omit_unset_slots=omit_unset_slots
+        )
+
+
+    def training_states_labels_and_entities(
+        self,
+        trackers: List[DialogueStateTracker],
+        domain: Domain,
+        omit_unset_slots: bool = False,
+    ) -> Tuple[List[List[State]], List[List[Text]], List[List[Dict[Text, Any]]]]:
+        """Transforms list of trackers to lists of states, labels and entity data.
+
+        Args:
+            trackers: The trackers to transform
+            domain: The domain
+            omit_unset_slots: If `True` do not include the initial values of slots.
+
+        Returns:
+            A tuple of list of states, list of labels and list of entity data.
         """
-        trackers_as_states = []
-        trackers_as_actions = []
-        trackers_as_entities = []
 
-        # from multiple states that create equal featurizations
-        # we only need to keep one.
-        hashed_examples = set()
+        self._setup_example_iterator()
+
+        example_states = []
+        example_labels = []
+        example_entities = []
 
         logger.debug(
-            "Creating states and action examples from "
+            "Creating states and {} label examples from "
             "collected trackers (by {}({}))..."
-            "".format(type(self).__name__, type(self.state_featurizer).__name__)
+            "".format(
+                self.LABEL_NAME,
+                type(self).__name__, 
+                type(self.state_featurizer).__name__
+            )
         )
         pbar = tqdm(
             trackers,
             desc="Processed trackers",
             disable=rasa.shared.utils.io.is_logging_disabled(),
         )
         for tracker in pbar:
-            states = self._create_states(
-                tracker, domain, omit_unset_slots=omit_unset_slots
-            )
 
-            states_length_for_action = 0
-            entity_data = {}
-            for event in tracker.applied_events():
-                if isinstance(event, UserUttered):
-                    entity_data = self._entity_data(event)
+            for states, label, entities in self._example_iterator(
+                tracker,
+                domain,
+                omit_unset_slots=omit_unset_slots
+            ):
 
-                if not isinstance(event, ActionExecuted):
+                if self._check_example_cache(tracker, states, label):
```

Comment:
Can we give this a more meaningful name, like `_check_duplicate_example`?

Reply:
Calling it `_is_duplicate_example`